packages/hdwallet-native/src/native.ts

```diff
@@ -62,11 +64,12 @@ class NativeHDWalletInfo extends MixinNativeBTCWalletInfo(MixinNativeETHWalletIn
   }
 }
 
-export class NativeHDWallet extends MixinNativeBTCWallet(MixinNativeETHWallet(NativeHDWalletInfo))
+export class NativeHDWallet
+  extends MixinNativeBTCWallet(MixinNativeETHWallet(MixinNativeCosmosWalletInfo(NativeHDWalletInfo)))
   implements core.HDWallet {
```

Comment:
I think we should require the implements for each mixin wallet as well to ensure interfaces are implemented correctly. `implements core.HDwallet, core.BTCWallet, core.ETHWallet, core.CosmosWallet`

Reply:
This would also have caught the incorrect functions signature for `cosmosSignTx`

Reply:
think we got this